app/main.py

```diff
@@ -1,3 +1,55 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int | float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Distance) -> int:
+        if isinstance(other, int | float):
+            return Distance(self.km + other)
+        return Distance(self.km + other.km)
+
+    def __iadd__(self, other: Distance) -> int:
+        if isinstance(other, int | float):
+            self.km += other
+            return self
+        self.km += other.km
+        return self
+
+    def __mul__(self, other: int | float) -> int:
+        return Distance(self.km * other)
+
+    def __truediv__(self, other: int | float) -> float:
+        return Distance(round(self.km / other, 2))
+
+    def __lt__(self, other: Distance) -> bool:
+        if isinstance(other, int | float):
+            return self.km < other
+        return self.km < other.km
+
+    def __eq__(self, other: Distance) -> bool:
+        if isinstance(other, int | float):
+            return self.km == other
+        return self.km == other.km
+
+    def __le__(self, other: Distance) -> bool:
```

Comment:
You can use previously created magic method here(and in `ge` as well) to simplify your code. e.g. `return not self > other`

Reply:
I'm sorry, but I don't quite understand what you mean.

Reply:
e.g. here in `le` we can use previously created magic methods. `return not self > other` if it's not bigger then it's lower or equal, and you don't need to check types, because `gt` will be triggered